dive_core/command_hierarchy.cpp

```diff
@@ -2275,35 +2241,30 @@ void CommandHierarchyCreator::AddSharedChild(CommandHierarchy::TopologyType type
 }
 
 //--------------------------------------------------------------------------------------------------
-// Remove all children listed in |children_node_indices|.
-void CommandHierarchyCreator::RemoveListOfChildren(
-CommandHierarchy::TopologyType type,
-uint64_t                       node_index,
-const std::vector<uint64_t>   &children_node_indices)
+void CommandHierarchyCreator::SetStartSharedChildrenNodeIndex(CommandHierarchy::TopologyType type,
+                                                              uint64_t node_index,
+                                                              uint64_t shared_child_node_index)
 {
-    if (children_node_indices.empty())
-    {
-        return;
-    }
-
-    auto  &vec = m_node_children[type][0][node_index];
-    size_t j = 0;
-    size_t k = 0;
-    for (size_t i = 0; i < vec.size(); i++)
-    {
-        if (j < children_node_indices.size() && vec[i] == children_node_indices[j])
-        {
-            j++;
-        }
-        else
-        {
-            vec[k++] = vec[i];
-        }
-    }
+    DIVE_ASSERT(node_index < m_node_start_shared_child[type].size());
+    m_node_start_shared_child[type][node_index] = shared_child_node_index;
+}
 
-    DIVE_ASSERT(j == children_node_indices.size());
+//--------------------------------------------------------------------------------------------------
+void CommandHierarchyCreator::SetEndSharedChildrenNodeIndex(CommandHierarchy::TopologyType type,
+                                                            uint64_t node_index,
+                                                            uint64_t shared_child_node_index)
+{
+    DIVE_ASSERT(node_index < m_node_end_shared_child[type].size());
+    m_node_end_shared_child[type][node_index] = shared_child_node_index;
+}
 
-    vec.erase(vec.begin() + vec.size() - children_node_indices.size(), vec.end());
+//--------------------------------------------------------------------------------------------------
+void CommandHierarchyCreator::SetSharedChildRootNodeIndex(CommandHierarchy::TopologyType type,
+                                                          uint64_t                       node_index,
+                                                          uint64_t root_node_index)
```

Comment:
is the root node always the submit node? could it be the marker node as well?

Reply:
It's explicitly set as the submit node in all calls to this function. If there's a topology where the submit doesn't show up, then it can be set to something else as well. But at the moment that's the only use case